src/main/java/com/bookclub/dao/BookDAO.java

```diff
@@ -12,6 +12,14 @@
 
 public class BookDAO implements IBookAO {
 
+    private Book buildBookFromResultSet(ResultSet resultSet) throws Exception {
```

Comment:
Throws exception? Why did no other function need to declare this for creating a book?

```suggestion
    private Book buildBookFromResultSet(ResultSet resultSet) {
```